src/main/java/com/google/cloud/tools/managedcloudsdk/internal/download/StandardDownloader.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.managedcloudsdk.internal.download;
+
+import com.google.common.annotations.VisibleForTesting;
+import java.io.BufferedOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.net.URLConnection;
+import java.nio.file.FileAlreadyExistsException;
+import java.nio.file.Files;
+import java.nio.file.NotDirectoryException;
+import java.nio.file.Path;
+
+/** Standard implementation of {@link Downloader}. */
+public class StandardDownloader implements Downloader {
+
+  static final int BUFFER_SIZE = 10 * 1024;
+  private final URL address;
+  private final Path destinationFile;
+  private final String userAgentString;
+  private final DownloadProgressListener downloadProgressListener;
+
+  /** Use {@link DownloaderFactory} to instantiate. */
+  StandardDownloader(
+      URL source,
+      Path destinationFile,
+      String userAgentString,
+      DownloadProgressListener downloadProgressListener) {
+    this.address = source;
+    this.destinationFile = destinationFile;
+    this.userAgentString = userAgentString;
+    this.downloadProgressListener = downloadProgressListener;
+  }
+
+  @Override
+  public Path call() throws IOException, InterruptedException {
+
+    createDestinationDirectory();
+    return downloadUrl();
+  }
+
+  @VisibleForTesting
```

Comment:
Typically you should test via public APIs and don't create backdoors specifically for tests: http://go/unit-test-practices#public-apis

Since nobody will call these methods in isolation (they will always use `call()`, consider making these methods private and only invoke `call()` in your tests.